modules/decklinksrc/decklinksrc.py

```diff
@@ -28,28 +31,40 @@ class DeckLinkSRC:
         Displays live video feed of DeckLink stream
     """
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         """
         Initializes DeckLink video stream
         """
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture('decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink') #Starts capture on initialization of object
+        self.capture = cv2.VideoCapture(
+            'decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink')  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        mainProcess = mp.Process(target=self._mainProcess_, daemon=True)
+        mainProcess.start()
+
+    def _mainProcess_(self):
+        print("decklink started")
+        while True:
+            newFrame = self.grab()
+            if self.videoPipeline.empty() or newFrame != self.videoPipeline.get():
+                self.videoPipeline.put(newFrame)
+            sleep(0.1)
 
     def stop(self):
         """
         Logic for stopping video feed by releasing capture and destroying any windows open.
         """
         self.capture.release()
         cv2.destroyAllWindows()
-    
 
-    def grab(self, frame):
+    def grab(self):  # Logic for capturing single frame from deckLink
```

Comment:
Why did you add this? It's repeated in the line below

Reply:
I'm also starting to wonder whether this function really serves any purpose; I can't think of many use cases where we would want to display a single frame

Reply:
This should just be to grab a single frame for processing, not display? The use case is grabbing a single frame for processing in targetAcquisition